File: files/CSEC 17.11.py
import pandas as pd

path = "files/Excel Challenge October 17th.xlsx"
input = pd.read_excel(path, usecols="B:D", skiprows=1, nrows=15)
test = pd.read_excel(path, usecols="E", skiprows=1, nrows=15)

input['group'] = (input['Defects'] != input['Defects'].shift()).cumsum()
input['Running Totals'] = input.groupby('group')['Units Made'].cumsum().where(input['Defects'] != 1, 0)

print(input['Running Totals'].eq(test['Running Totals']).all()) # True

File: files/CSEC 17.11.R
library(tidyverse)
library(readxl)

path = "files/Excel Challenge October 17th.xlsx"
input = read_excel(path, range = "B2:D16")
test  = read_excel(path, range = "E2:E16")

result = input %>%
  mutate(group = consecutive_id(Defects)) %>%
  mutate(`Running Totals` = ifelse(Defects != 1, cumsum(`Units Made`), 0), .by = group)
         
all.equal(result$`Running Totals`, test$`Running Totals`)
# [1] TRUE